File: backend/app/src/main/java/backend/model/entity/EntityResponse.java
package backend.model.entity;

import backend.model.PairResponse;
import backend.model.player.PlayerResponse;

public abstract class EntityResponse {
    public final String id;

    public PairResponse position;

    public int hp;

    public PlayerResponse owner;

    public EntityResponse(Entity entity) {
        this.id = entity.id;
        this.position = new PairResponse(entity.position);
        this.hp = entity.hp;
        this.owner = new PlayerResponse(entity.owner);
    }
}

File: frontend/src/ts/lib/model/EntityResponse.ts
import PlayerResponse from "./PlayerResponse.js";
import Position from "./Position.js";

export default abstract class EntityResponse {
  public id: string;

  public position: Position;

  public hp: number;

  public owner: PlayerResponse;

  constructor(
    id: string,
    position: Position,
    hp: number,
    owner: PlayerResponse
  ) {
    this.id = id;
    this.position = position;
    this.hp = hp;
    this.owner = owner;
  }
}
